core/tee/tee_pobj.c

```diff
@@ -37,20 +37,21 @@ TAILQ_HEAD_INITIALIZER(tee_pobjs);
 static TEE_Result tee_pobj_check_access(uint32_t oflags, uint32_t nflags)
 {
 	/* meta is exclusive */
-	if (oflags | TEE_DATA_FLAG_ACCESS_WRITE_META ||
-	    nflags | TEE_DATA_FLAG_ACCESS_WRITE_META)
-		return TEE_ERROR_ACCESS_CONFLICT;
-
-	if (oflags | TEE_DATA_FLAG_ACCESS_READ &&
-	    !((nflags | TEE_DATA_FLAG_SHARE_READ) &&
-	      oflags | TEE_DATA_FLAG_SHARE_READ))
-		return TEE_ERROR_ACCESS_CONFLICT;
-
-	if (oflags | TEE_DATA_FLAG_ACCESS_WRITE &&
-	    !((nflags | TEE_DATA_FLAG_SHARE_WRITE) &&
-	      oflags | TEE_DATA_FLAG_SHARE_WRITE))
+	if (oflags & TEE_DATA_FLAG_ACCESS_WRITE_META)
 		return TEE_ERROR_ACCESS_CONFLICT;
 
+	/*second open/create need set flags to TEE_DATA_FLAG_SHARE_READ */
+	if (((oflags & TEE_DATA_FLAG_ACCESS_READ) ||
```

Comment:
Maybe It's likely  to be incorrect.
As a example, oflags is 0/NULL and nflags is TEE_DATA_FLAG_SHARE_READ, in your above code, tee_pobj_check_access  will return SUCCESS.
Table 5-14, TEE_OpenPersistentObject Sharing Rules in GP core internal API 1.1, last line said:"An object can be opened with no flag set, which completely locks all subsequent attempts to access the object. Only the first call will succeed.", so in this case, return TEE_ERROR_ACCESS_CONFLICT would be right.
So I suggest to use:
if (!((nflags & TEE_DATA_FLAG_SHARE_READ) &&
      (oflags & TEE_DATA_FLAG_SHARE_READ)))
    return TEE_ERROR_ACCESS_CONFLICT;
to instead your code.
It means  existed persistent object's sharing rules flags need to set  SHARE to allow other object handle to share read/write this object(Sharing permission control flags,page 119,GP core api 1.1). And  other shared  object handle need to set SHARE way to use(5.7.3 Persistent Object Sharing Rules,GP core api 1.1).
Do you agree with me?